### PR TITLE
doc: document pythonRelaxDepsHook

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Description of changes

Documentation of `pythonRelaxDepsHook`, introduced in https://github.com/NixOS/nixpkgs/pull/170813.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
CC @symphorien.

File: doc/languages-frameworks/python.section.md [hunk 1]
```diff
@@ -663,6 +663,68 @@ However, this is done in it's own phase, and not dependent on whether `doCheck =
...
+```
+pkg1<1.0
+pkg2
+pkg3>=1.0,<=2.0
+```
+
+we can do:
+
+```
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
```
Comment (thiagokokada) [inline doc/languages-frameworks/python.section.md:690]: Eventually this hook should be included by default in `buildPythonPackage`, but for now it needs to be included manually.

File: doc/languages-frameworks/python.section.md [hunk 2]
```diff
@@ -663,6 +663,70 @@ However, this is done in it's own phase, and not dependent on whether `doCheck =
 This can also be useful in verifying that the package doesn't assume commonly
 present packages (e.g. `setuptools`)
 
+#### Using pythonRelaxDepsHook {#using-pythonrelaxdepshook}
+
+It is common for upstream to specifies a range of versions for its package
```
Comment (symphorien) [inline doc/languages-frameworks/python.section.md:668]:
```suggestion
It is common for upstream to specify a range of versions for its package
```

File: doc/languages-frameworks/python.section.md [hunk 3]
```diff
@@ -663,6 +663,70 @@ However, this is done in it's own phase, and not dependent on whether `doCheck =
...
+dependencies. This makes sense, since it ensures that the package will be build
```
Comment (symphorien) [inline doc/languages-frameworks/python.section.md:669]:
```suggestion
dependencies. This makes sense, since it ensures that the package will be built
```

**Review by symphorien — APPROVED** [review]
lgtm apart from two minor typos

**Pull request merged**